src/groups/rotation_action.jl

```diff
@@ -304,3 +304,108 @@ function optimal_alignment(
     Ostar = det(UVt) ≥ 0 ? UVt : F.U * Diagonal([i < L ? 1 : -1 for i in 1:L]) * F.Vt
     return convert(typeof(Xmul), Ostar)
 end
+
+@doc raw"""
+    ComplexPlanarRotation()
+
+Action of the circle group [`CircleGroup`](@ref) on ``ℝ^2`` by left multiplication.
+"""
+struct ComplexPlanarRotation <: AbstractGroupAction{LeftAction} end
+
+base_group(::ComplexPlanarRotation) = CircleGroup()
+
+group_manifold(::ComplexPlanarRotation) = Euclidean(2)
+
+@doc raw"""
+    apply(A::ComplexPlanarRotation, g::Complex, p)
+
+Rotate point `p` from [`Euclidean(2)`](@ref) manifold by the group element `g`.
+The formula reads
+````math
+p_{rot} =  \begin{bmatrix}
+\cos(θ) & \sin(θ)\\
+-\sin(θ) & \cos(θ)
+\end{bmatrix} p,
+````
+where `θ` is the argument of complex number `g`.
+"""
+function apply(::ComplexPlanarRotation, g::Complex, p)
+    sinθ, cosθ = g.im, g.re
+    return (@SMatrix [cosθ -sinθ; sinθ cosθ]) * p
+end
+apply(::ComplexPlanarRotation, ::Identity{MultiplicationOperation}, p) = p
+
+function apply!(A::ComplexPlanarRotation, q, g::Complex, p)
+    return copyto!(q, apply(A, g, p))
+end
+function apply!(::ComplexPlanarRotation, q, ::Identity{MultiplicationOperation}, p)
+    return copyto!(q, p)
+end
+
+@doc raw"""
+    QuaternionRotation
+
+Action of the unit quaternion group [`UnitaryMatrices`](@ref)`(1, ℍ)` on ``ℝ^3``.
```

Comment:
Does that mean, that we missed that unitary matrices already make #259 a bit obsolete? Because I just gave that task to students to get a bit into the code base here

Reply:
How is #259 related to this?

Reply:
To exactly zero percent. I copied https://github.com/JuliaManifolds/Manifolds.jl/issues/529 – and have no clue how that got mixed up.

Reply:
Ah, no, #529 is a superset of what I've done here.

Reply:
Ok, perfect. We might get that somewhen from two students in Heidelberg, if all works well.